src/infrastructure/TransactionSearchGroup.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A transaction could be classified in the following groups: * Unconfirmed: The transaction reached the P2P network. At this point, it is not guaranteed that the transaction will be included in a block. * Confirmed: The transaction is included in a block. * Partial: The transaction requires to be cosigned by other transaction participants in order to be included in a block. * Failed: The transaction did not pass the network validation, and it was rejected.
+ */
+export enum TransactionSearchGroup {
```

Comment:
Can we rename it to `TransactionGroup`? It's going to be used in more ways than just search endpoint.

The way I see it is: 
TransactionGroup = confirmed | partial | unconfirmed
TransactionState =  confirmed | partial | unconfirmed | error

I've also added an optional group to the transaction in Java to be filled depending when how the transaction is loaded. If the transaction is loaded using the `confirmed` search, `confirmed` listener or `confirmed` by id, the group is `confirmed`. The state/group is not going to be added to the rest payload but it's still good that the user has the state/group at hand

https://github.com/nemtech/symbol-openapi/issues/159

https://github.com/nemtech/symbol-sdk-java/blob/961/sdk-core/src/main/java/io/nem/symbol/sdk/model/transaction/Transaction.java#L279